src/utils/format-numbers.js

```diff
@@ -4,38 +4,42 @@
 
 // @flow
 
+import type { Microseconds, Milliseconds } from '../types/units';
+
 /**
  * Format a positive float into a string.
  *
- * Try to format the value to 2 significant digits as much as possible but
- * without using scientific notation.  The number of decimal places depends
- * on the value: the closer to zero the value is, the more decimal places
- * are used in the resulting string.
+ * Try to format the value to num_digits significant digits as much as
+ * possible but without using scientific notation.  The number of decimal
+ * places depends on the value: the closer to zero the value is, the more
+ * decimal places are used in the resulting string.  No more than max_places
+ * decimal places will be used.
  *
- * For example:
+ * For example, using num_digits = 2 (the default):
  *
  * formatNumber(123,   ) = "123"
  * formatNumber(12.3,  ) =  "12"
  * formatNumber(1.23,  ) =   "1.2"
  * formatNumber(0.01234) =   "0.012"
  */
-export function formatNumber(value: number): string {
-  let result: string;
-  if (value >= 10) {
-    result = value.toFixed(0);
-  } else if (value >= 1) {
-    result = value.toFixed(1);
-  } else if (value >= 0.1) {
-    result = value.toFixed(2);
-  } else {
-    result = value.toFixed(3);
+export function formatNumber(
+  value: number,
+  num_digits: number = 2,
+  max_places: number = 3
+): string {
+  const num_digits_on_left = Math.floor(Math.log10(value)) + 1;
```

Comment:
> nit 3: I know this is painful to move between style guides, but our variable names are 
> camelCased. Only constants use _ on our codebase, unless we have to access some external 
> APIs.

I'm sorry I keep forgetting which coding standard I'm using.  The Gecko people don't like me either <joking>, since SpiderMonkey is also different from Gecko.